test/e2e/framework/framework.go

```diff
@@ -117,7 +118,7 @@ func New(kubeConfigPath string) (*Framework, CleanUpFunc, error) {
 		return nil, nil, fmt.Errorf("creating monitoring beta client failed: %w", err)
 	}
 
-	operatorClient, err := client.NewForConfig(config, "", namespaceName, userWorkloadNamespaceName)
+	operatorClient, err := client.NewForConfig(config, "", namespaceName, userWorkloadNamespaceName, client.EventRecorder(events.NewInMemoryRecorder("cluster-monitoring-operator")))
```

Comment:
Minor improvement for future readers: Maybe leave a small comment as to what the event recorder does for us here?

Reply:
sure, I'll add that.
Maybe to [pkg/client/client_test.go](https://github.com/openshift/cluster-monitoring-operator/pull/2226/files#diff-2dad64e28ce5e2eddf0adf303052d38e3b5b4fcba6b32fc4ca26d5e4d6b8fa4a) as well (talking about it incase you missed it as it's collapsed by default)